apps/api/src/app/controllers/edgeConnection.controller.ts

```diff
@@ -0,0 +1,133 @@
+import httpStatusCodes from '../constants/httpStatusCodes';
+
+@ApiTags('connections')
+@Controller('connections')
+export class EdgeConnectionController {
+    constructor(private readonly repositoryProvider: RepositoryProvider) {}
+
+    @Get('')
+    async getStats() {
+        const count = await this.repositoryProvider.getEdgeConnectionRepository().getCount();
+
+        return {
+            count,
+        };
+    }
+
+    @Get('notes')
+    async fetchManyNotes(@Res() res) {
+        const result = await this.repositoryProvider.getEdgeConnectionRepository().fetchMany();
+
+        const errors = result.filter((result) => isInternalError(result));
+
+        if (result.some(isInternalError))
+            return res.status(httpStatusCodes.internalError).send({
+                errors,
+            });
+
+        const edgeConnections = result as EdgeConnection[];
+
+        const noteViewModels = edgeConnections.map(
+            (edgeConnection) => new NoteViewModel(edgeConnection)
+        );
+
+        return res.status(httpStatusCodes.ok).send(noteViewModels.map(cloneToPlainObject));
+    }
+
+    @Get('selfNotes')
+    async fetchResourceSelfConnections(
+        @Res() res,
+        // be careful, these are actually unknown but Swagger uses this type for doc generation
+        @Query('id') id: string,
+        @Query('type') type: string
+    ) {
+        if (!isResourceId(id))
+            return res
+                .status(httpStatusCodes.badRequest)
+                .send(new InternalError(`Invalid resource id: ${id}`));
+
+        if (!isResourceType(type))
+            return res
+                .status(httpStatusCodes.badRequest)
+                .send(new InternalError(`Invalid resource type: ${type}`));
+
+        const result = await this.repositoryProvider.getEdgeConnectionRepository().fetchMany();
+
+        const errors = result.filter(isInternalError);
+
+        if (errors.length) return res.status(httpStatusCodes.badRequest).send({ errors });
+
+        const allEdgeConnections = result as EdgeConnection[];
+
+        const selfEdgeConnectionsForThisResource = allEdgeConnections.filter(
+            ({ type: edgeConnectionType, members }) =>
+                edgeConnectionType === EdgeConnectionType.self &&
+                isDeepStrictEqual(members[0].compositeIdentifier, { type, id })
+        );
+
+        res.status(httpStatusCodes.ok).send(
+            cloneToPlainObject(
+                selfEdgeConnectionsForThisResource.map(
+                    (connection) => new NoteViewModel(connection)
+                )
+            )
+        );
+    }
+
+    @Get('forResource')
+    async fetchConnectionsForResource(
+        @Res() res,
+        // be careful, these are actually unknown but Swagger uses this type for doc generation
+        @Query('id') id: string,
+        @Query('type') type: string
+    ) {
+        if (!isResourceId(id))
```

Comment:
Could this be refactored since it's used above?

Reply:
Yes, it **should** be refactored!. Maybe into `validateResourceCompositeIdFromQueryParams`?

Reply:
I've added a story for this [here](https://www.pivotaltracker.com/story/show/182052280).